src/test/groovy/transform/stc/GenericsSTCTest.groovy

```diff
@@ -1800,6 +1800,19 @@ assert result == 'ok'
 '''
     }
 
+    // GROOVY-7742
+    void testMethodParamWithDifferentPlaceholderForTypeArgument() {
+        // We use <T> here while List uses <E>
```

Comment:
That was my thought initially as well, but in that case I would expect the following to fail since AtomicReference uses 'V' as a placeholder:
```
import java.util.concurrent.atomic.AtomicReference

@groovy.transform.CompileStatic
class Foo {
    public <V> AtomicReference<V> firstReference(List<AtomicReference<V>> references) {
        return references.first()
    }
}
```

The code above compiles but, interestingly enough, the variant below does not:

```
import java.util.concurrent.atomic.AtomicReference                                 

@groovy.transform.CompileStatic                                                                                   
class Foo {                                                                        
    public <T> AtomicReference<T> firstClass(List<AtomicReference<T>> references) {
        return references.first()                                                  
    }                                                                              
}
```